presto-main/src/main/java/io/prestosql/dispatcher/FailedDispatchQuery.java

```diff
@@ -104,9 +104,9 @@ public Session getSession()
     }
 
     @Override
-    public ListenableFuture<?> getDispatchedFuture()
+    public ListenableFuture<DispatchStatus> getDispatchedFuture()
     {
-        return immediateFuture(null);
+        return immediateFuture(FAILED);
```

Comment:
don't use static import for `DispatchStatus`. `QueryState` seems to be more commonly used.

Reply:
You can import both. But I can use fully qualified for DispatchStatus if you prefer